zenoh-plugin-ros2dds/Cargo.toml

```diff
@@ -43,13 +44,15 @@ futures = { workspace = true }
 git-version = { workspace = true }
 hex = { workspace = true }
 lazy_static = { workspace = true }
+r2r = { version = "0.9", optional = true }
```

Comment:
The version number must be set in wokspace's `Cargot.toml` as for other deps.

Also, it must be a `dev-dependencies` as used only for tests.

Reply:
Unfortunately, we can't have `optional` inside `dev-dependencies`
https://github.com/rust-lang/cargo/issues/1596

The reason why I use `optional` here is that rust-analyzer will show warning if the developer didn't source ROS 2 environment due to `r2r`. To avoid this, the `r2r` will only be used if we want to run CI test related to ROS (enable `ros_test` feature)

Reply:
Fair enough! I just added comments in `Cargo.toml` so we don't forgot this detail.